jbi/models.py

```diff
@@ -304,11 +306,15 @@ def extract_from_see_also(self):
 
     def lookup_action(self, actions: Actions) -> Action:
         """Find first matching action from bug's whiteboard list"""
-        tags: list[str] = self.get_potential_whiteboard_config_list()
-        for tag in tags:
-            if action := actions.get(tag):
-                return action
-        raise ActionNotFoundError(", ".join(tags))
+        if self.whiteboard:
+            for tag, action in actions.by_tag.items():
+                pre = r"\[.*" if action.brackets_required else ""
+                post = r".*\]" if action.brackets_required else ""
+                search_string = f"{pre}{tag.lower()}{post}"
+                if re.search(search_string, self.whiteboard.lower()):
```

Comment:
nitpick: instead of turning both inputs to lowercase, you can do: `if re.search(search_string, self.whiteboard, flags=re.IGNORE_CASE):`